File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i], map.getOrDefault(nums[i],0)+1);
        }
        PriorityQueue<Integer> minheap = new PriorityQueue<>();
        for(int key:map.keySet()){
            minheap.add(key);
        }
        int[] ans = new int[k];
        for(int i=0;i<k;i++){
            ans[i] = minheap.poll();
        }
        return ans;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        for num in nums:
            freq[num] = freq.get(num,0)+1
        heap = [(-count, num) for num, count in freq.items()]
        heapq.heapify(heap)
        ans = []
        for _ in range(k):
            count, num = heapq.heappop(heap)
            ans.append(num)
        return ans